docs/Artifacts definition format and style guide.asciidoc

```diff
@@ -535,5 +571,16 @@ supported_os: [Windows, Linux, Darwin]
 
 == [[parameter_expansion]]Parameter expansion and globs
 
-*TODO*
+Path, keys, key and query attributes can contain parameter expansion and globing. This allows
+for flexible creation of artifact locations.
+
+Parameter expansions values are enclosed by double percent symbols e.g. `%%environ_systemroot%%`.
+The parameter expansion value can be replaced by the corresponding value from the
+<<knowledge_base,knowledge base>>.
 
+Parameter can also contain regular glob elements (such as `**`, `*`, `?`, `[a-z]`). For
+example, having files `foo`, `bar`, `baz` glob expansion of `ba?` will yield
+`bar` and `baz`. Group expansion allows defining lists of possible artifact locations for example,
+given path `foo/{bar,baz}/{quux,norf}` the locations `foo/bar/quux`, `foo/bar/norf`, `foo/baz/quux`
+and `foo/baz/norf` are defined. A recursive component (specified as `**`) matches any directory
```

Comment:
These globstar defaults same to be based on the GRR implementation. WIP to flush out the details https://github.com/ForensicArtifacts/artifacts/issues/342